### PR TITLE
Issue #139 Loading Banner in lazy loading

### LINKED ISSUE #139: Banner should not be visible before other page items
The banner becomes visible before other items on the page.  It should become visible along with the whole page.

Here's a video capturing the problem on n/w throttled to slow 3G 

https://user-images.githubusercontent.com/1191451/228002467-c34ad24b-dd3c-4f4f-a6c9-7e057bbf9a9f.mov

### PR DESCRIPTION
**Pull request by jindaliiita**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #139

Test URLs:
- Before: https://main--nedbank--hlxsites.hlx.live/content/nedbank/za/en/personal/home
- After: https://issue-139-new--nedbank--hlxsites.hlx.live/content/nedbank/za/en/personal/home

### CODE REVIEW

**Review by sdmcraft — COMMENTED** [review]
https://user-images.githubusercontent.com/1191451/229285162-ff12ad19-cac4-4f0b-93bb-382f3eebefe7.mov

This seems to be introducing a CLS where the image is pushed down after the banner appears (as can also be seen in the PSI audit). The banner placeholder should keep occupying its space  even before it renders to avoid this hero image getting pushed down. I think the prior code ensured that but also ended up showing the contents too soon. What we need is a placeholder that occupies space and gets filled up asynchronously with the actual banner later on when the whole page becomes visible.

**Comment by jindaliiita** [comment]
> This seems to be introducing a CLS where the image is pushed down after the banner appears (as can also be seen in the PSI audit). The banner placeholder should keep occupying its space even before it renders to avoid this hero image getting pushed down. I think the prior code ensured that but also ended up showing the contents too soon. What we need is a placeholder that occupies space and gets filled up asynchronously with the actual banner later on when the whole page becomes visible.

Yes, After pushing the code i figured out this problem. That's why i have not added anyone for review :laughing:

File: blocks/banner/banner.js [hunk 1]
```diff
@@ -8,12 +12,29 @@ function setCookie(name, value, timeInMillis, path) {
...
   setCookie('oldSitePopUpCookies', true, 86400000, '/');
 }
 
-export default function decorate(block) {
+export default async function decorate(block) {
+  const resp = await fetch(`${window.hlx.codeBasePath}${getRootPath()}/banner.plain.html`);
+
+  if (resp && resp.status === 200) {
+    const section = document.querySelector('.banner-placeholder.section');
+    section.style['background-color'] = '#eeeeee';
```
Comment (vdua) [inline blocks/banner/banner.js:25]: can we not have a class to do that instead of inline-style ?

File: blocks/banner/banner.js [hunk 2]
```diff
@@ -8,12 +12,29 @@ function setCookie(name, value, timeInMillis, path) {
 
 function handleCloseButtonClick() {
   const banner = document.querySelector('.banner-placeholder');
-  banner.classList.remove('appear');
+  banner.classList.add('hide');
```
Comment (vdua) [inline blocks/banner/banner.js:15]: where are we removing this class ?

File: blocks/banner/banner.js (hunk 1 above)
Comment (jindaliiita) [reply blocks/banner/banner.js]: Done

**Review by sdmcraft — APPROVED** [review]

**Pull request merged**